internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -52,70 +124,152 @@ func (c *ClusterManagerImpl) Stop() error {
 	return make([]dcache.RawVolume, 0)
 }
 
-// IsAlive implements ClusterManager.
-func (c *ClusterManagerImpl) IsAlive(nodeId string) bool {
+// isAlive implements ClusterManager.
+func (c *ClusterManagerImpl) isAlive(nodeId string) bool {
 	return false
 }
 
-// LowestNumberRV implements ClusterManager.
-func (c *ClusterManagerImpl) LowestNumberRV(rvNames []string) []string {
+// lowestNumberRV implements ClusterManager.
+func (c *ClusterManagerImpl) lowestNumberRV(rvNames []string) []string {
 	return make([]string, 0)
 }
 
-// NodeIdToIP implements ClusterManager.
-func (c *ClusterManagerImpl) NodeIdToIP(nodeId string) string {
+// nodeIdToIP implements ClusterManager.
+func (c *ClusterManagerImpl) nodeIdToIP(nodeId string) string {
 	return ""
 }
 
-// RVFsidToName implements ClusterManager.
-func (c *ClusterManagerImpl) RVFsidToName(rvFsid string) string {
+// rVFsidToName implements ClusterManager.
+func (c *ClusterManagerImpl) rVFsidToName(rvFsid string) string {
 	return ""
 }
 
-// RVNameToFsid implements ClusterManager.
-func (c *ClusterManagerImpl) RVNameToFsid(rvName string) string {
+// rVNameToFsid implements ClusterManager.
+func (c *ClusterManagerImpl) rVNameToFsid(rvName string) string {
 	return ""
 }
 
-// RVNameToIp implements ClusterManager.
-func (c *ClusterManagerImpl) RVNameToIp(rvName string) string {
+// rVNameToIp implements ClusterManager.
+func (c *ClusterManagerImpl) rVNameToIp(rvName string) string {
 	return ""
 }
 
-// RVNameToNodeId implements ClusterManager.
-func (c *ClusterManagerImpl) RVNameToNodeId(rvName string) string {
+// rVNameToNodeId implements ClusterManager.
+func (c *ClusterManagerImpl) rVNameToNodeId(rvName string) string {
 	return ""
 }
 
-// ReportRVDown implements ClusterManager.
-func (c *ClusterManagerImpl) ReportRVDown(rvName string) error {
+// reportRVDown implements ClusterManager.
+func (c *ClusterManagerImpl) reportRVDown(rvName string) error {
+	return nil
+}
+
+// reportRVFull implements ClusterManager.
+func (c *ClusterManagerImpl) reportRVFull(rvName string) error {
 	return nil
 }
 
-// ReportRVFull implements ClusterManager.
-func (c *ClusterManagerImpl) ReportRVFull(rvName string) error {
+func (cmi *ClusterManagerImpl) checkAndCreateInitialClusterMap(dCacheConfig *dcache.DCacheConfig, rvList []dcache.RawVolume) error {
+	isClusterMapExists, err := cmi.checkIfClusterMapExists()
+	if err != nil {
+		log.Err("ClusterManagerImpl::checkAndCreateInitialClusterMap: Failed to check clusterMap file presence in Storage. error -: %v", err)
+		return err
+	}
+	if isClusterMapExists {
+		log.Trace("ClusterManager::checkAndCreateInitialClusterMap : ClusterMap.json already exists")
+		return nil
+	}
+	currentTime := time.Now().Unix()
+	clusterMap := dcache.ClusterMap{
+		Readonly:      evaluateReadOnlyState(),
+		State:         dcache.StateReady,
+		Epoch:         1,
+		CreatedAt:     currentTime,
+		LastUpdatedAt: currentTime,
+		LastUpdatedBy: rvList[0].NodeId,
```

Comment:
as commented in the punch heartbeat PR, nodeId is a global property so best stored in the ClusterManagerImpl.
Then we should not pass rvList to checkAndCreateInitialClusterMap, as it's non-intuitive. Initial clustermap creation should not have anything to do with this node's rvlist.